File: chapters/chapter2/scripts/matlab/GaussQuad.m
function [d,w] = GaussQuad(n,s0,s1)
  if nargin==1
    s0=-1; s1=1; 
  end
  if n==1
    d=(s0+s1)/2; w=s1-s0; return; 
  end
  j=1:n-1; bta=j./sqrt(4*j.^2-1); Q=diag(bta,-1)+diag(bta,1);
  [U,L]=eig(Q);
  [d,k]=sort(diag(L));
  w=2*U(1,k).^2; w=w(:);
  d=s0+0.5*(s1-s0)*(d+1);
  w=0.5*(s1-s0)*w;
end

File: chapters/chapter2/scripts/python/GaussQuad.py
def GaussQuad(n,s0,s1):
    nargin=len(locals())
    d=[]; w=[]
    if nargin==1:
        s0=-1; s1=1
    if n==1:
        d=(s0+s1)/2; w=s1-s0  
        return d,w 
    j=numpy.arange(1,n)
    bta=j/numpy.sqrt(4*(j**2)-1)   
    Q=numpy.diag(bta, k=-1)+numpy.diag(bta, k=1)
    L,U = numpy.linalg.eig(Q)
    k=numpy.argsort(L)
    d=L[k]; w=2*(U[0][k]**2)
    d=s0+0.5*(s1-s0)*(d+1)
    w=0.5*(s1-s0)*w
    return d,w